pkg/registries/set_impl.go

```diff
@@ -49,6 +50,27 @@ func (e *setImpl) GetAll() []types.ImageRegistry {
 	return e.getSortedRegistriesNoLock()
 }
 
+// GetAllUnique returns the unique set of integrations that are active.
+// Unique integrations are dependent on the integrations URL and auth information.
+func (e *setImpl) GetAllUnique() []types.ImageRegistry {
+	e.lock.RLock()
+	defer e.lock.RUnlock()
+	// Get the integrations sorted first. This ensures that we dedupe deterministically with the downside that
+	// we iterate over the list of integrations twice.
+	integrations := e.getSortedRegistriesNoLock()
+
+	uniqueKeys := set.NewStringSet()
+	uniqueIntegrations := make([]types.ImageRegistry, 0, len(integrations))
+	for _, i := range e.integrations {
```

Comment:
Should we only de-dupe autogenerated?

Thinking for manual integrations that is the 'users choice' and we should respect that if they have dupes, however if an autogenerated integration matches an manual integration, omit it - I assume the bulk of issues we've seen were due to the mass amount of auto-generated.

Reply:
Yeah it makes sense, the autogenerated ones hurt us the most for the time being, I'd be surprised if there's customers who just create high cardinality duplicates.